Pyjion/pyjit.cpp

```diff
@@ -75,20 +108,43 @@ extern "C" __declspec(dllexport) PyJittedCode* JitCompile(PyCodeObject* code) {
 #ifdef DEBUG_TRACE
         printf("Compilation failure #%d\r\n", ++failCount);
 #endif
-        return nullptr;
-    }
-
-    auto jittedCode = (PyJittedCode*)PyJittedCode_New();
-    if (jittedCode == nullptr) {
-        // OOM
-        delete res;
-        return nullptr;
+        jittedCode->j_failed = true;
+        return false;
     }
 
-    g_jittedCode[jittedCode] = res;
+    g_pyjionJittedCode[jittedCode] = res;
     jittedCode->j_evalfunc = &Jit_EvalHelper;
     jittedCode->j_evalstate = res->get_code_addr();
-    return jittedCode;
+    return true;
+}
+
+static PY_UINT64_T HOT_CODE = 20000;
+
+extern "C" __declspec(dllexport) PyObject *EvalFrame(PyFrameObject *f, int throwflag) {
```

Comment:
As discussed elsewhere privately, we do if we want to follow my idea of simply assigning `PyEval_EvalFrameEx()` to the slot and generalizing this even farther.